bundle/src/main/java/com/adobe/acs/commons/redirects/filter/RedirectFilter.java

```diff
@@ -0,0 +1,508 @@
+                        String queryString = slingRequest.getQueryString();
+                        if (queryString != null) {
+                            int idx = location.indexOf('?');
+                            if (idx == -1) idx = location.indexOf("#");
+                            if (idx != -1) location = location.substring(0, idx);
+
+                            location += "?" + queryString;
+                        }
+                    }
+
+                    if(urlAdjuster != null){
+                        location = urlAdjuster.adjust(slingRequest, location);
+                    }
+                }
+
+                log.info("Redirecting {} to {}, statusCode: {}",
+                        resourcePath, location, redirectRule.getStatusCode());
+                slingResponse.setHeader("Location", location);
+                for(String name : onDeliveryHeaders.keySet()){
+                    String value = onDeliveryHeaders.get(name);
+                    slingResponse.setHeader(name, value);
+                }
+                slingResponse.setStatus(redirectRule.getStatusCode());
+                redirected = true;
+            }
+        }
+        return redirected;
+    }
+
+    String rewriteUrl(String url, ResourceResolver resourceResolver) {
+        return resourceResolver.map(url);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    protected boolean rewriteUrls() {
+        return rewriteUrls;
+    }
+
+    /**
+     * @return whether redirection is enabled
+     */
+    protected boolean isEnabled() {
+        return enabled;
+    }
+
+    public String getStoragePath() {
+        return storagePath;
+    }
+
+    protected Collection<String> getExtensions() {
+        return exts;
+    }
+
+    protected Collection<String> getPaths() {
+        return paths;
+    }
+
+    protected Collection<String> getMethods() {
+        return methods;
+    }
+
+    /**
+     * Check whether redirection for the given request is allowed.
+     * <ol>
+     * <li>On author redirects are disabled in EDIT, PREVIEW and DESIGN WCM Modes.
+     * To test on author you need to disable WCM mode and append &wcmmode=disabled
+     * to the query string</li>
+     * <li>Redirects are supported only for GET and HEAD methods</li>
+     * This can be changed in the OSGi configuration</li>
+     * <li>If configured, redirects are allowed only for the specified extensions,
+     * e.g. only *.html requests will be redirected. Same path with .json extension
+     * will <i>not</i> be redirected. This feature is disabled by default.</li>
+     * </ol>
+     *
+     * @param request the request to check
+     * @return whether redirection for the given is allowed
+     */
+    private boolean doesRequestMatch(SlingHttpServletRequest request) {
+        WCMMode wcmMode = WCMMode.fromRequest(request);
+        if (WCMMode.EDIT == wcmMode || WCMMode.PREVIEW == wcmMode || WCMMode.DESIGN == wcmMode) {

```

Comment:
I think it would be more stable to just check for `WCMMode.DISABLED != wcmMode`

Reply:
Changed